File: fix_rules.js
import fs from 'fs';
let content = fs.readFileSync('firestore.rules', 'utf8');

content = content.replace(
  /match \/monthlyRevenue\/\{id\} \{\n\s*allow read: if isStaff\(\);\n\s*allow write: if isStaff\(\);\n\s*allow delete: if isAdmin\(\);\n\s*\}/,
  `match /monthlyRevenue/{id} {
      allow read: if isStaff();
      allow write: if isStaff();
      allow delete: if isAdmin();
    }

    match /yearlyRevenue/{id} {
      allow read: if isStaff();
      allow write: if isStaff();
      allow delete: if isAdmin();
    }`
);

fs.writeFileSync('firestore.rules', content);

File: fix_rules.py
import re

with open('firestore.rules', 'r') as f:
    rules = f.read()

old_rule = "(isStaff() && incoming().diff(existing()).affectedKeys().hasAny(['waiterId', 'waiterName', 'waiterStatus', 'status', 'kitchenStatus', 'barmanStatus', 'isPaid', 'paymentMethod', 'updatedAt', 'verificationToken', 'deliveredAt', 'paidAt', 'deliveredInMinutes', 'readyInMinutes', 'readyAt', 'preparingAt', 'completedAt', 'kitchenStartedAt', 'kitchenReadyAt', 'kitchenCompletedAt', 'barmanStartedAt', 'barmanReadyAt', 'barmanCompletedAt']));"
new_rule = "(isStaff());"

rules = rules.replace(old_rule, new_rule)

with open('firestore.rules', 'w') as f:
    f.write(rules)

print("Rules updated.")
